src/main/java/com/oltpbenchmark/api/Worker.java

```diff
@@ -483,6 +556,42 @@ protected final void doWork(DatabaseType databaseType, TransactionType transacti
 
     }
 
+    /**
+     * Checks to see if the exception indicates that the current connection is read-only.
+     * @param ex
+     * @return
+     */
+    private boolean indicatesReadOnly(SQLException ex) {
+
+        String sqlState = ex.getSQLState();
+        int errorCode = ex.getErrorCode();
+
+        LOG.debug("sql state [{}] and error code [{}]", sqlState, errorCode);
```

Comment:
- [ ] Wrap in `LOG.isDebugEnabled` check

Reply:
i thought best practice these days was to forgo the isDebugEnabled type checks.   i'm fine for you to leave that out

Reply:
Yeah, I guess for Java it doesn't matter here.  In other places we want to avoid the expensive string formatting operations in case the call isn't necessary.